global_helpers/panther_base_helpers.py

```diff
@@ -107,6 +107,15 @@ def gsuite_parameter_lookup(parameters, key):
     return None
 
 
+def gsuite_details_lookup(detail_type, detail_names, event):
+    for details in event.get('events', {}):
+        if (details.get('type') == detail_type and
+                details.get('name') in detail_names):
+            return details
+    # not found, return empty dict
+    return {}
```

Comment:
If we are ok with this helper, I can go ahead and convert all the gsuite rules that do this type of looping.  Currently, I only updated the ones where I updated the titles.

Reply:
This helper looks good, and sort of matches the above one. I would suggest adding a comment though explaining what it does and why.

If you want to standardize on using this function in all the gsuite rules I say go for it.